service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -75,9 +87,38 @@ public final void setTracer(final Tracer opentracingTracer) {
      */
     @Override
     public final void start(final Future<Void> startFuture) {
+
+        if (vertx != null) {
+            vertx.setPeriodic(HEARTBEAT_RATE_MILIIS, id -> updateLiveness("heartbeat", Status.OK()));
+        }
+
         startInternal().setHandler(startFuture.completer());
     }
 
+    protected void updateReadiness(final String procedureName, final Status status) {
+        sendHealthStatus(procedureName, status, HealthCheckServer.EVENT_BUS_ADDRESS_HEALTH_READINESS);
+    }
+
+    protected void updateLiveness(final String procedureName, final Status status) {
+        sendHealthStatus(procedureName, status, HealthCheckServer.EVENT_BUS_ADDRESS_HEALTH_LIVENESS);
+    }
+
+    protected final void sendHealthStatus(final String procedureName, final Status status, final String address) {
+        final Instant now = Instant.now();
+
+        final DeliveryOptions options = new DeliveryOptions()
+                .addHeader(HealthCheckServer.MSG_FIELD_VERTICLE_NAME, this.getClass().getName())
+                .addHeader(HealthCheckServer.MSG_FIELD_VERTICLE_INSTANCE_ID, instanceID.toString())
+                .addHeader(HealthCheckServer.MSG_FIELD_PROCEDURE_NAME, procedureName)
+                .addHeader(HealthCheckServer.MSG_FIELD_LAST_SEEN, String.valueOf(now.getEpochSecond()));
+
+        status.getData().put(HealthCheckServer.MSG_FIELD_LAST_SEEN, now.toString());
```

Comment:
Yes, I like to have the time as a readable string in the `Status` because I found it very convenient when I manually checked the readiness or liveness probes. The epoch timestamp in the header is evaluated for the liveness check by the health check server. But it is of course redundant and I understand that we want to be resource efficient. So we can drop one of them.

Reply:
> But it is of course redundant and I understand that we want to be resource efficient. So we can drop one of them.

I think we can afford the few extra bytes. The *readable* last seen will then also be reported to the external monitoring tool as part of the HTTP response body, right?

Reply:
Yes.